hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -861,6 +875,12 @@ private boolean checkUserError(int responseStatusCode) {
         && responseStatusCode < HttpURLConnection.HTTP_INTERNAL_ERROR);
   }
 
+  private boolean isMd5ChecksumError(final AzureBlobFileSystemException e) {
```

Comment:
if you make this static and package private you can add a unit test to verify that it is handled

+add javadocs

Reply:
Added test to verify that this works and we return proper Exceptions to the caller in case of MD5Mismatch.
Not sure if we still want to make this static.

Reply:
static? just because its easier to test standalone. if you don't need that, don't worry